src/psyclone/tests/profile_test.py

```diff
@@ -0,0 +1,547 @@
+            {1}[type='inner',field_space='cv',it_space='internal_pts']
+                {2} compute_cv_code(cv_fld,p_fld,v_fld) '''
+        '''[module_inline=False]
+        {1}[type='outer',field_space='ct',it_space='all_pts']
+            {1}[type='inner',field_space='ct',it_space='all_pts']
+                {2} bc_ssh_code(ncycle,p_fld,tmask) '''
+        '''[module_inline=False]'''.format(coloured_schedule, coloured_loop,
+                                           coloured_kern, coloured_profile)
+    )
+    assert correct in out
+
+    prt = ProfileRegionTrans()
+
+    # Insert a profile call between outer and inner loop.
+    # This is tests that we find the subroutine node even
+    # if it is not the immediate parent.
+    new_sched, _ = prt.apply(invoke.schedule.children[0]
+                             .children[0].children[0])
+
+    new_sched_str = str(new_sched)
+    correct = ("""GOSchedule(Constant loop bounds=True):
+ProfileStart[]
+Loop[]: j= lower=2,jstop-1,1
+ProfileStart[]
+Loop[]: i= lower=2,istop,1
+kern call: compute_cv_code
+EndLoop
+ProfileEnd
+EndLoop
+Loop[]: j= lower=1,jstop+1,1
+Loop[]: i= lower=1,istop+1,1
+kern call: bc_ssh_code
+EndLoop
+EndLoop
+ProfileEnd
+End Schedule""")
+
+    assert correct in new_sched_str
+
+    Profiler.set_options(None)
+
+
+# -----------------------------------------------------------------------------
+def test_profile_errors2():
+    '''Test various error handling.'''
+
+    with pytest.raises(GenerationError) as gen_error:
+        Profiler.set_options(["invalid"])
+    assert "Invalid option" in str(gen_error)
+
+
+# -----------------------------------------------------------------------------
+def test_profile_invokes_gocean1p0():
+    '''Check that an invoke is instrumented correctly
+    '''
+    Profiler.set_options([Profiler.INVOKES])
+    _, invoke = get_invoke("gocean1.0", "test11_different_iterates_over_"
+                           "one_invoke.f90", 0)
+
+    # Convert the invoke to code, and remove all new lines, to make
+    # regex matching easier
+    code = str(invoke.gen()).replace("\n", "")
+
+    # First a simple test to that the nesting is correct - the
+    # profile regions include both loops
+    correct_re = ("subroutine invoke.*"
+                  "use profile_mod, only: ProfileData.*"
+                  r"TYPE\(ProfileData\), save :: profile.*"
+                  "call ProfileStart.*"
+                  "do j.*"
+                  "do i.*"
+                  "call.*"
+                  "end.*"
+                  "end.*"
+                  "call ProfileEnd")
+    assert re.search(correct_re, code, re.I) is not None
+
+    _, invoke = get_invoke("gocean1.0", "single_invoke_"
+                           "two_kernels.f90", 0)
+
+    # Convert the invoke to code, and remove all new lines, to make
+    # regex matching easier
+    code = str(invoke.gen()).replace("\n", "")
+
+    correct_re = ("subroutine invoke.*"
```

Comment:
Please add a comment describing what this test is for. Please also extend the test to check that the arguments to the profiling-calls are correct.